### PR TITLE
Add multipart request sending option

### PR DESCRIPTION
**Pull request by dukaarpad**
According to https://www.w3.org/TR/SOAP-attachments

### CODE REVIEW

**Comment by TheKidCoder** [comment]
This is great! Any update on when we could get a merge?

**Comment by dukaarpad** [comment]
I'm still working on it, because the Mail module was changed in meantime.
I'll send a new pull-request soon.

2017-05-31 22:59 GMT+02:00 Chris Ostrowski <notifications@github.com>:

**Comment by dukaarpad** [comment]
The Mail::Patterns reference is removed (it was omitted from Mail).
Missing tests added and fixed.

**Comment by AgeevAndrew** [comment]
:+1: 

In my humble opinion, it is an utterly important thing. Are there any reasons, why this request can't be merged?
I write one gem with dependency on savon, and I need this functionality in savon gem.

**Comment by shirkavand** [comment]
Guys any update on this?

**Comment by dukaarpad** [comment]
Hi, I made tests, cleaned the code and rebased with actual version. I cannot do more.

**Comment by Xosmond** [comment]
Heeeey, @dukaarpad can I use ur fork meanwhile this PR is accepted?

**Comment by Xosmond** [comment]
Please guys, this is an important PR.

**Comment by dukaarpad** [comment]
Hi,

Yes, of course you may fork it. I'll keep it while it will be pulled in
main gem, because I already have around 15 rails applications in
production, which uses this gem with these features.

And also, if you have some further fixes or developments, feel free to code
it an send a pull request.

Best,
Duka

2017-10-31 0:11 GMT+01:00 Jordano Moscoso <notifications@github.com>:

File: README.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ $ gem install savon
 or add it to your Gemfile like this:
 
 ```
-gem 'savon', '~> 2.11.1'
+gem 'savon', '~> 2.12.0'
```
Comment (pcai) [inline README.md]: can you please revert this change? This version is not yet released.

**Comment by pcai** [comment]
Apologies for the lack of movement here. The reason I have not merged is because it looks like a large change, including introducing new dependency, and haven't had much time to dig into it. 

I looked at it briefly just now, over next few days I will try to do a more detailed pass. Thanks everyone for your patience.

**Comment by dukaarpad** [comment]
Please check it deeply and tell if you see problems. And even if you found something which could be solved in a better or prettier way.

**Comment by webler-duka** [comment]
We are still using this feature and would be good if it will be merged in main branch.

**Comment by dukaarpad** [comment]
Please do not close this. It's alive and we are still waiting to be merged in main repo.

**Comment by pcai** [comment]
Thanks for your contribution and for your patience.

**Comment by dukaarpad** [comment]
Thanks for your help.

Regards,
Duka Árpád

2018-04-29 18:34 GMT+02:00 Peter Cai <notifications@github.com>:

**Comment by Erpizzo72** [comment]
Anyone have a complete sample for request with attachments?

**Comment by guihatano** [comment]
If am I right, it is not on 2.12.0
I tried to use master but I didn't have success.

As @Erpizzo72 says, someone could help us on how to implement it?

**Comment by Piioo** [comment]
Hi,

is this working somehow with SOAP/XOP https://www.w3.org/TR/xop10/ because this change has broke this functionality in connection with the https://github.com/savonrb/savon-multipart gem

How I make `<xop:Include xmlns:xop='http://www.w3.org/2004/08/xop/include'  href='cid:http://example.org/me.png'/>` to bring back to work, that the data is added into the XML?

**Comment by rubiii** [comment]
Hey everyone. With v2.17.2, attachments now work with a user-supplied xml envelope.
Hope that fixes this problem. Let me know if you still have any problems.
